File: hash_tool/hash_tool.cpp
#include <iostream>
#include <string>
#include <cstdlib>
using namespace std;

void hashString(const string& input, const string& algorithm) {
    string command;

    if (algorithm == "md5") {
        command = "echo -n \"" + input + "\" | openssl md5";
    } else if (algorithm == "sha256") {
        command = "echo -n \"" + input + "\" | openssl sha256";
    } else {
        cout << "Unsupported algorithm." << endl;
        return;
    }

    system(command.c_str());
}

int main() {
    string text;
    int choice;

    cout << "Hash Generator (C++)\n" << endl;
    cout << "Enter text to hash: ";
    getline(cin, text);

    cout << "\nChoose algorithm:\n1. MD5\n2. SHA-256\nChoice: ";
    cin >> choice;

    if (choice == 1) {
        cout << "\nMD5 Hash:\n";
        hashString(text, "md5");
    } else if (choice == 2) {
        cout << "\nSHA-256 Hash:\n";
        hashString(text, "sha256");
    } else {
        cout << "Invalid option." << endl;
    }

    return 0;
}

File: hash_tool/hash_tool.py
import hashlib

def hash_string(data, algorithm='sha256'):
    if algorithm == 'md5':
        return hashlib.md5(data.encode()).hexdigest()
    elif algorithm == 'sha256':
        return hashlib.sha256(data.encode()).hexdigest()
    else:
        return None

if __name__ == "__main__":
    print("Hash Generator (Python)\n")
    text = input("Enter text to hash: ")

    print("\nChoose algorithm:")
    print("1. MD5")
    print("2. SHA-256")
    choice = input("Enter 1 or 2: ")

    if choice == '1':
        result = hash_string(text, 'md5')
        print("\nMD5 Hash:")
    elif choice == '2':
        result = hash_string(text, 'sha256')
        print("\nSHA-256 Hash:")
    else:
        print("\nInvalid choice.")
        exit()

    print(result)
